### PR TITLE
add ui 

### PR DESCRIPTION
**Pull request by jian-he**

### CODE REVIEW

File: console/backend/pkg/handlers/notebook.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+	klog.Infof("notebook notebook %v ", notebook)
+	if err == nil {
+		return nh.client.Create(context.Background(), notebook)
+	} else {
+		// fallback to yaml format
+		err = yaml.Unmarshal(data, notebook)
+		if err != nil {
+			klog.Errorf("failed to unmarshal notebook in yaml format, data: %s", string(data))
+			return err
+		}
```
Comment (SimonCqk) [inline console/backend/pkg/handlers/notebook.go:58]: create a notebook instance after yaml marshaling check passed ?

File: console/backend/pkg/model/converter.go [hunk 2]
```diff
@@ -112,7 +125,40 @@ func GetTimeDiffer(startTime time.Time, endTime time.Time) (differ string) {
...
+	now := time.Now()
+	if !dmoNotebook.GmtCreated.IsZero() {
+		if !util.Time(dmoNotebook.GmtTerminated).IsZero() {
+			// already terminated
+			notebookInfo.DurationTime = GetTimeDiffer(dmoNotebook.GmtCreated, *dmoNotebook.GmtTerminated)
+		} else {
+			notebookInfo.DurationTime = GetTimeDiffer(dmoNotebook.GmtCreated, now)
+		}
+	}
+	notebookInfo.UserName = *dmoNotebook.Owner
```
Comment (SimonCqk) [inline console/backend/pkg/model/converter.go:161]: `dmoNotebook.Owner` can be a nil value ?

File: console/backend/pkg/model/notebook.go [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+	// status of Notebook
+	NotebookStatus string `json:"notebookStatus"`
+
+	// url to the notebook
+	Url string `json:"url"`
+
+	// Namespace
+	Namespace string `json:"namespace"`
+	//
```
Comment (SimonCqk) [inline console/backend/pkg/model/notebook.go:28]: remove `//` or complete the comments

File: console/backend/pkg/model/workspace.go [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+// WorkspaceInfo is the object returned in http call
+type WorkspaceInfo struct {
+	Username string `json:"username"`
+
+	Namespace string `json:"namespace"`
+
+	CPU int64 `json:"cpu"`
+
+	Memory int64 `json:"memory"`
```
Comment (SimonCqk) [inline console/backend/pkg/model/workspace.go:14]: what is the unit of cpu/memory/gpu value ?

File: pkg/storage/backends/objects/mysql/mysql.go [hunk 5]
```diff
@@ -410,6 +551,18 @@ func (b *mysqlBackend) updateJob(oldJob, newJob *dmo.Job) error {
 	return nil
 }
 
+func (b *mysqlBackend) CreateWorkspace(workspace *model.WorkspaceInfo) error {
+	panic("implement me")
```
Comment (SimonCqk) [inline pkg/storage/backends/objects/mysql/mysql.go:555]: make it a no-op function instead of just panic ?

File: pkg/tensorboard/tensorboard.go [hunk 6]
```diff
@@ -283,16 +282,16 @@ func syncIngress(jc job_controller.JobController, c client.Client,
...
 
 	ingSpec := opts.Ingress
-	if ingSpec == nil || ingSpec.Host == nil || len(validation.IsDNS1123Subdomain(*ingSpec.Host)) > 0 {
-		return nil
-	}
+	//if ingSpec == nil  {
+	//	return nil
+	//}
 	pathPrefix := ""
 	if ingSpec.PathPrefix != nil {
```
Comment (SimonCqk) [inline pkg/tensorboard/tensorboard.go:289]: ingSpec can be nil at this point

File: console/backend/pkg/handlers/notebook.go (hunk 1 above)
Comment (jian-he) [reply console/backend/pkg/handlers/notebook.go:58]: fixed

File: console/backend/pkg/model/converter.go (hunk 2 above)
Comment (jian-he) [reply console/backend/pkg/model/converter.go:161]: maybe, is this a problem ？

File: console/backend/pkg/model/notebook.go (hunk 3 above)
Comment (jian-he) [reply console/backend/pkg/model/notebook.go:28]: removed, the filed name is self-explanatory enough

File: console/backend/pkg/model/workspace.go (hunk 4 above)
Comment (jian-he) [reply console/backend/pkg/model/workspace.go:14]: this is not implemented yet...

File: console/backend/pkg/model/converter.go (hunk 2 above)
Comment (SimonCqk) [reply console/backend/pkg/model/converter.go:161]: then dereference `*dmoNotebook.Owner` may panic

File: console/backend/pkg/storage/objects/apiserver/apiserver.go [hunk 7]
```diff
@@ -265,6 +272,300 @@ func (a *apiServerBackend) DeleteJob(ns, name, jobID, kind, region string) error
...
+		}
+	}
+
+	pvc := &corev1.PersistentVolumeClaim{}
+	pvcName := types.NamespacedName{
+		Name:      model.WorkspacePrefix + namespace.Name,
+		Namespace: namespace.Namespace,
+	}
+
+	storageClassName := "my-azurefile"
```
[the remark this answers is not in the document]
Comment (jian-he) [reply console/backend/pkg/storage/objects/apiserver/apiserver.go:503]: this functionality is disabeld for now

**Pull request merged**